app/models/multi_pool.rb

```diff
@@ -0,0 +1,60 @@
+# frozen_string_literal: true
+
+# MultiPool
+# A collection of pools grouped/created together using a specific pooling method.
+class MultiPool < ApplicationRecord
+  include Pipelineable
+
+  enum :pool_method, { Plate: 0, TubeRack: 1 }
+
+  has_many :multi_pool_positions, dependent: :destroy
+
+  validates :pool_method, presence: true
+  validates :pipeline, presence: true
+  validates :multi_pool_positions, presence: true
+  validate :consistent_pools_type?
+  validate :unique_pool_positions?
+
+  accepts_nested_attributes_for :multi_pool_positions, allow_destroy: true
+
+  # Checks that all pools in the multi pool are of the same type.
+  # @return [void]
+  def consistent_pools_type?
+    return true if multi_pool_positions.empty?
+
+    # Check which pool types are present for each position
+    # We can't simply use pool_type because of the polymorphic association
+    # may have nil pool_type if the pool is not set e.g. during nested creation
+    types = multi_pool_positions.map do |pos|
```

Comment:
I would delegate this to the multi_pool_position model so you don't need a guard.

e.g. `types = multi_pool_positions.map(&:pipeline).compact.uniq`